lib/index.js

```diff
@@ -307,18 +352,28 @@ export function createUnifiedLanguageServer({
         /* c8 ignore next */
         if (!cwd) return
 
+        const configuration = await getDocumentSettings(textDocument.uri)
+
         const file = lspDocumentToVfile(textDocument, cwd)
 
-        const files = workspacePathToFiles.get(cwd) || []
-        workspacePathToFiles.set(cwd, [...files, file])
+        if (configuration.requireConfig) {
+          const files = workspacePathToFilesRequireConfig.get(cwd) || []
+          workspacePathToFilesRequireConfig.set(cwd, [...files, file])
+        } else {
+          const files = workspacePathToFiles.get(cwd) || []
+          workspacePathToFiles.set(cwd, [...files, file])
+        }
       })
     )
 
     /** @type {Array<Promise<Array<VFile>>>} */
     const promises = []
 
     for (const [cwd, files] of workspacePathToFiles) {
-      promises.push(processWorkspace(cwd, files, alwaysStringify))
+      promises.push(processWorkspace(cwd, files, alwaysStringify, false))
+    }
+    for (const [cwd, files] of workspacePathToFilesRequireConfig) {
+      promises.push(processWorkspace(cwd, files, alwaysStringify, true))
```

Comment:
- `processWorkspace` becomes hard to read with so many props, especially with a boolean somewhere. Should it accept an object?
- Also is quite close to duplicated code, perhaps that can be streamlined?

Reply:
not going to block it but I do definitely prefer the object style, in this case too. I’m trying to move everything, with very rare exceptions, to: a) either one object, b) main value and then an options object!

Reply:
I’m going to keep this as-is for now, but if one more option is added, I’ll change it to an options object.